sky/provision/azure/config.py

```diff
@@ -86,6 +86,8 @@ def bootstrap_instances(
                                   'use_external_resource_group field')
     use_external_resource_group = provider_config['use_external_resource_group']
 
+    subnet_id = provider_config.get('subnet_id', '')
```

Comment:
nit: can we use None instead of empty string to indicate not specified? same for the arm template.